File: faceblur/apps/backend/internal/domain/image.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type Image struct {
	ID          uuid.UUID
	URL         *string
	OriginalURL string
	CreatedBy   string
	CreatedAt   time.Time
	UpdatedAt   time.Time
	IsProcessed bool
	IsFailed    bool
}

type ImageProcessingQueue struct {
	ID            uuid.UUID
	ImageID       uuid.UUID
	CreatedBy     string
	CreatedAt     time.Time
	UpdatedAt     time.Time
	CompletedAt   *time.Time
	IsFailed      bool
	FailureReason *string
	OperationID   *string
}

type ListImagesRequest struct {
	PageNavigation PageNavigationRequest
}

type ListImagesResponse struct {
	Images         []Image
	PageNavigation PageNavigationResponse
}

File: faceblur/apps/dashboard/src/models/image.ts
export type ImageModel = {
	id: string;
	url: string;
	isProcessed: boolean;
	isFailed: boolean;
	createdAt: Date | null;
	updatedAt: Date | null;
};

export type Upload = {
	id: string;
	file?: File;
	uploadedAt: Date | null;
	image?: ImageModel;
	failed?: boolean;
	progress: number;
};
